### PR TITLE
dm-99-write-use-case/extend-existing-if-it-makes-sense-to-get-all-sample-objects-for-a-project-code

### PR DESCRIPTION
**Pull request by KochTobi**
Many thanks for contributing to this project!

**PR Checklist**
Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

- [x] This comment contains a description of changes (with reason)
- [x] Referenced issue is linked
- [x] If you've fixed a bug or added code that should be tested, add tests!

**Description of changes**
This PR addresses the use case. It adds the possibility to query for sample information for a given project.

### CODE REVIEW

File: sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/download/DownloadSamplesDataSource.groovy [hunk 1]
```diff
@@ -20,4 +20,11 @@ interface DownloadSamplesDataSource {
      * @since 1.0.0
      */
     List<String> fetchSampleCodesFor(String projectCode, Status status)
+
+    /**
+     * Given a project code, returns all sample codes with that project
+     * @param projectCode
+     * @return a list of sample codes for the given project
+     */
+    List<String> fetchSampleCodesFor(String projectCode)
```
Comment (jenniferboedker) [inline sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/download/DownloadSamplesDataSource.groovy:29]: Warum ist das in DownloadSamplesDataSource? Gehört das nicht in GetSampleInfo?

Comment (jenniferboedker) [reply sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/download/DownloadSamplesDataSource.groovy:29]: Would it make sense to throw a DataSourceException?

File: sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/info/SampleStatusDataSource.groovy [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import life.qbic.datamodel.samples.Status
+
+/**
+ * <b>Provides information on samples and their status</b>
+ *
+ *
+ * @since 1.0.0
+ */
+interface SampleStatusDataSource {
+    Map<String, Status> fetchSampleStatusesFor(Collection<String> sampleCodes)
```
Comment (jenniferboedker) [inline sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/info/SampleStatusDataSource.groovy:12]: Maybe add a javadoc for this method :)

**Review by jenniferboedker — CHANGES_REQUESTED** [review]
Overall looks good, with only some minor questions that need to be resolved.

**Comment by KochTobi** [comment]
Hi @jenniferboedker Thank you for your review, I addressed your concerns.

File: sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/info/SampleStatusDataSource.groovy [hunk 3]
```diff
@@ -9,5 +10,13 @@ import life.qbic.datamodel.samples.Status
  * @since 1.0.0
  */
 interface SampleStatusDataSource {
-    Map<String, Status> fetchSampleStatusesFor(Collection<String> sampleCodes)
+
+    /**
+     * Retrieves sample statuses for a collection of provided sample codes
+     * @param sampleCodes the sample codes of samples that should be checked for their sample status
+     * @return a map containing statueses for every sample code as key
```
Comment (jenniferboedker) [inline sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/info/SampleStatusDataSource.groovy:17]:
```suggestion
     * @return a map containing statuses for every sample code as key
```

**Review by jenniferboedker — APPROVED** [review]

**Pull request merged**